src/compiler/emitter.ts

```diff
@@ -3012,6 +3012,26 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                 return result;
             }
 
+            function emitEs6ExportDefaultCompat(node: Node) {
+                if (node.parent.kind === SyntaxKind.SourceFile) {

```

Comment:
@JsonFreeman I think this line is required. if this line changes into `Debug.assert`, we got a assertion error in wrong context.

```
  3 failing

  1) compiler tests for tests/cases/compiler/moduleElementsInWrongContext.ts "before all" hook:
     Error: Debug Failure. False expression:
      at Object.assert (eval at <anonymous> (built/local/run.js:52497:13), <anonymous>:1533:23)

  2) compiler tests for tests/cases/compiler/moduleElementsInWrongContext2.ts "before all" hook:
     Error: Debug Failure. False expression:
      at Object.assert (eval at <anonymous> (built/local/run.js:52497:13), <anonymous>:1533:23)

  3) compiler tests for tests/cases/compiler/moduleElementsInWrongContext3.ts "before all" hook:
     Error: Debug Failure. False expression:
      at Object.assert (eval at <anonymous> (built/local/run.js:52497:13), <anonymous>:1533:23)
```